bql/planner/tracer/trace.go

```diff
@@ -25,19 +25,32 @@ type Arguments struct {
 	Msgs []string
 }
 
+// event encapsulates a single tracing event.
 type event struct {
 	w          io.Writer
 	t          time.Time
 	tracerArgs func() *Arguments
 }
 
-var c chan *event
+// MessageTracer encapsulates the intrinsic verbosity of a given tracing message.
+type MessageTracer struct {
+	verbosity int
+}
+
+// tracerVerbosity represents the global verbosity level of the current tracer. Level 1 means minimum verbosity (printing
+// only what is crucial) while level 3 means maximum verbosity (printing all available tracing messages).
+var tracerVerbosity int
```

Comment:
nit: did you consider using iota for the data type?

Iota is the "golang enum". Iota represents successive untypes integer constants which seems what you need here PTAL: https://golang.org/ref/spec#Iota

By giving a name to the integers we can also improve the readability and instead of SetVerbosity(1) do a SetVerbosity(PRINT_ONLY) or MINIMUM, wdyt?

Reply:
Thank you for the comment Jose. I do use `iota` in other places of the code (for the `FILTER` function types in `filter.go`, for example), but for the verbosity levels above we discussed a little about and, to follow the internal code that is often using `.V(1)` (with the explicit `int` level being passed), we decided to continue passing the explicit `int` level as well, for consistency.

Since the comments of the tracing methods are already being explicit about the semantics of these levels, it should not be a problem for a developer looking at the code to understand them. Also, the current way is a bit more concise if we compare to something like `.V(tracer.MINIMUM)` before each call to `.Trace`.